draft-reddy-add-resolver-info-06.txt

```diff
@@ -0,0 +1,448 @@
+   "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+   "OPTIONAL" in this document are to be interpreted as described in BCP
+   14 [RFC2119][RFC8174] when, and only when, they appear in all
+   capitals, as shown here.
+
+
+
+
+Reddy & Boucadair         Expires 3 March 2023                  [Page 2]
+
+Internet-Draft          DNS Resolver Information             August 2022
+
+
+   This document makes use of the terms defined in [RFC8499].
+
+   'Encrypted DNS' refers to a DNS protocol that provides an encrypted
+   channel between a DNS client and server (e.g., DoT, DoH, or DoQ).
+
+3.  Retrieving Resolver Information
+
+   A stub resolver that wants to retrieve the resolver information may
+   use the RRtype "RESINFO" defined in this document (see Section 7.1).
+
+   The content of the RDATA in a response to RRtype query is defined in
+   Section 5.  If the resolver understands the RESINFO RRtype, the RRset
+   in the Answer section MUST have exactly one record.
+
+   The client can retrieve the resolver information using the RESINFO
+   RRtype and QNAME of the domain name that is used to authenticate the
+   DNS server (referred to as ADN in [I-D.ietf-add-dnr]).
+
+   If the special use domain name "resolver.arpa" defined in
+   [I-D.ietf-add-ddr] is used to discover the Encrypted DNS server, the
+   client can retrieve the resolver information using the RESINFO RRtype
+   and QNAME of the designated resolver.
+
+4.  Format of the Resolver Information
+
+   The resolver information uses the same format as DNS TXT records.
+   The intention of using the same format as TXT records is to convey a
+   small amount of useful information about a resolver.  The format
+   rules for TXT records are defined in Section 3.3.14 of the DNS
+   specification [RFC1035] and Section 6.1 of the DNS-based Service
+   Discovery [RFC6763].  The recommendations to limit the TXT record
+   size are discussed in Section 6.2 of [RFC6763].  The RESINFO RRtype
+   uses key/value pairs to convey the resolver information.  Each key/
+   value pair is encoded using the format rules defined in Section 6.3
+   of DNS-SD [RFC6763].  Using standardized key/value syntax within the
+   RESINFO RRtype makes it easier for additional keys to be defined
+   later.  If an implementation sees unknown keys in a RESINFO RRtype,
+   it MUST silently ignore them.  The rules for the keys is defined in
+   Section 6.4 of [RFC6763].
+
+   All keys MUST either be defined in the IANA registry Section 7.2 or
+   begin with the substring "temp-" for names defined for local use
+   only.
+
+5.  Resolver Information
+
+   The resolver information includes the following keys and values:
+
+
+
+
+Reddy & Boucadair         Expires 3 March 2023                  [Page 3]
+
+Internet-Draft          DNS Resolver Information             August 2022
+
+
+   qnamemin:  If the DNS server supports QNAME minimisation [RFC9156] to
```

Comment:
```suggestion
   qnamemin key:  If the DNS server supports QNAME minimisation [RFC9156] to
```

Reply:
I don't see the need to add the keyword "key".